FULL-feat-creditCardEditModal 결재수단 관리 모달 UI 수정 및 관련 API 구현

JunYoung7 opened the pull request:
### 📕 Issue Number

Linked #66 
<br/>

### 📗 작업 내역

> 구현 내용 및 작업 했던 내역

- FE
   - 결재수단 관리 UI 수정
![image](https://user-images.githubusercontent.com/61405355/101807272-46727880-3b58-11eb-9eab-17b646fda5c4.png)

   - 입력 폼을 이용하여 값을 받아오고, 버튼에 클릭 이벤트로 axios 요청

- BE
   - POST 결재수단 API 함수 생성
![image](https://user-images.githubusercontent.com/61405355/101807145-1f1bab80-3b58-11eb-8737-cc9c1ce7cc6d.png)

<br/>

### 📘 작업 유형

- 신규 기능 추가
- 버그 수정
- 리펙토링

<br/>

### 📝 PR 특이 사항

> PR을 볼 때 주의깊게 봐야하거나 말하고 싶은 점

- payment에 추가하기 전에 payment에 똑같은 이름이 있다면 추가하지 않고 user_payment에만 추가해줄 수 있도록 구현하였습니다.

<br/><br/>

JunYoung7 committed:
feat : 결재 수단 이름 입력 변수 설정

- 결재 수단 이름의 state를 관리하기 위해 useState 사용
- 기존의 line atom을 사용하지 말고 input atom을 사용하도록 변경

JunYoung7 committed:
refactor : ColumnFlexContainer z-index 속성 추가


# <Why>를 위한 <What>
# 로그인을 위한 OAuth api 구현
# 객체 의존성 역전을 위한 프로토콜 추상화

# <타입> 리스트
#   feat    : 기능 (새로운 기능)
#   fix     : 버그 (버그 수정)
#   refactor: 리팩토링
#   style   : 스타일 (코드 형식, 세미콜론 추가: 비즈니스 로직에 변경 없음)
#   docs    : 문서 (문서 추가, 수정, 삭제)
#   test    : 테스트 (테스트 코드 추가, 수정, 삭제: 비즈니스 로직에 변경 없음)
#   chore   : 기타 변경사항 (빌드 스크립트 수정 등)
# --------------------------------------------------------------
# 메세지는 한글로 작성
# 타입 첫 문자는 소문자로 작성
# 제목의 끝은 명사형으로 작성
# 제목 끝에 마침표(.) 금지
# 본문은 "어떻게" 보다 "무엇을", "왜"를 설명

JunYoung7 committed:
refactor : CreditCard 수정

- align을 위한 container들 추가

JunYoung7 committed:
refactor : CreditCardEditFormBox 수정

- name 전달을 위한 button event 등록

JunYoung7 committed:
refactor : CreditCardEditModal 수정

- 모달 내에서만 보여지기 위한 ScrollDiv 적용
- createButtonClick 이벤트 등록
- 따로 CreditCardEditCardList를 사용하지 않고 모달내에서 구현
# <Why>를 위한 <What>
# 로그인을 위한 OAuth api 구현
# 객체 의존성 역전을 위한 프로토콜 추상화

# <타입> 리스트
#   feat    : 기능 (새로운 기능)
#   fix     : 버그 (버그 수정)
#   refactor: 리팩토링
#   style   : 스타일 (코드 형식, 세미콜론 추가: 비즈니스 로직에 변경 없음)
#   docs    : 문서 (문서 추가, 수정, 삭제)
#   test    : 테스트 (테스트 코드 추가, 수정, 삭제: 비즈니스 로직에 변경 없음)
#   chore   : 기타 변경사항 (빌드 스크립트 수정 등)
# --------------------------------------------------------------
# 메세지는 한글로 작성
# 타입 첫 문자는 소문자로 작성
# 제목의 끝은 명사형으로 작성
# 제목 끝에 마침표(.) 금지
# 본문은 "어떻게" 보다 "무엇을", "왜"를 설명

JunYoung7 committed:
feat : payment 생성, 삭제 api 구현


# <Why>를 위한 <What>
# 로그인을 위한 OAuth api 구현
# 객체 의존성 역전을 위한 프로토콜 추상화

# <타입> 리스트
#   feat    : 기능 (새로운 기능)
#   fix     : 버그 (버그 수정)
#   refactor: 리팩토링
#   style   : 스타일 (코드 형식, 세미콜론 추가: 비즈니스 로직에 변경 없음)
#   docs    : 문서 (문서 추가, 수정, 삭제)
#   test    : 테스트 (테스트 코드 추가, 수정, 삭제: 비즈니스 로직에 변경 없음)
#   chore   : 기타 변경사항 (빌드 스크립트 수정 등)
# --------------------------------------------------------------
# 메세지는 한글로 작성
# 타입 첫 문자는 소문자로 작성
# 제목의 끝은 명사형으로 작성
# 제목 끝에 마침표(.) 금지
# 본문은 "어떻게" 보다 "무엇을", "왜"를 설명

JunYoung7 committed:
refactor : CreditCardEditCardList 수정

- organisms 미사용 예정으로 인한 파일 수정

JunYoung7 committed:
refactor : createButtonClick 이벤트 수정


생성 요청을 보내기 위한 AXIOS 함수 사용

JunYoung7 committed:
feat : payment, userpayment 관련 query 생성

- 생성, 조회 관련 query 생성

JunYoung7 committed:
feat : createUserPayments 구현 및 router 지정

- client api 형식에 맞추어 router 지정
- payments 중복 생성을 막기 위한 선 검사 후 payments와 userPayments에 생성해줄 수 있도록 구현
# <Why>를 위한 <What>
# 로그인을 위한 OAuth api 구현
# 객체 의존성 역전을 위한 프로토콜 추상화

# <타입> 리스트
#   feat    : 기능 (새로운 기능)
#   fix     : 버그 (버그 수정)
#   refactor: 리팩토링
#   style   : 스타일 (코드 형식, 세미콜론 추가: 비즈니스 로직에 변경 없음)
#   docs    : 문서 (문서 추가, 수정, 삭제)
#   test    : 테스트 (테스트 코드 추가, 수정, 삭제: 비즈니스 로직에 변경 없음)
#   chore   : 기타 변경사항 (빌드 스크립트 수정 등)
# --------------------------------------------------------------
# 메세지는 한글로 작성
# 타입 첫 문자는 소문자로 작성
# 제목의 끝은 명사형으로 작성
# 제목 끝에 마침표(.) 금지
# 본문은 "어떻게" 보다 "무엇을", "왜"를 설명

JunYoung7 committed:
feat : payment 생성, 조회 관련 service 구현


# <Why>를 위한 <What>
# 로그인을 위한 OAuth api 구현
# 객체 의존성 역전을 위한 프로토콜 추상화

# <타입> 리스트
#   feat    : 기능 (새로운 기능)
#   fix     : 버그 (버그 수정)
#   refactor: 리팩토링
#   style   : 스타일 (코드 형식, 세미콜론 추가: 비즈니스 로직에 변경 없음)
#   docs    : 문서 (문서 추가, 수정, 삭제)
#   test    : 테스트 (테스트 코드 추가, 수정, 삭제: 비즈니스 로직에 변경 없음)
#   chore   : 기타 변경사항 (빌드 스크립트 수정 등)
# --------------------------------------------------------------
# 메세지는 한글로 작성
# 타입 첫 문자는 소문자로 작성
# 제목의 끝은 명사형으로 작성
# 제목 끝에 마침표(.) 금지
# 본문은 "어떻게" 보다 "무엇을", "왜"를 설명

Zigje9 commented on client/src/components/organisms/CreditCardEditModal/CreditCardEditModal.tsx:
```diff
@@ -1,30 +1,40 @@
+
+    console.log(result);
+  };
+
+  const cardNameList = ['신한카드', '하나카드', '카카오페이', '신한카드', '하나카드', '카카오페이'];
+  const cardList = cardNameList.map((cardName) => <CreditCard name={cardName} />);
```
cardNameList 데이터가 중복되어있는데 의도하신건가요?

JunYoung7 replied:
바로 수정하겠습니다 !

leejiwoo2002 commented on server/src/payment/controller.ts:
```diff
@@ -3,10 +3,31 @@ import * as response from '../utils/response';
+  let checkInclude;
+  let newUserPaymentId;
+
+  for (const name of names) {
+    checkInclude = await Service.getPaymentIdByName(name);
+    if (checkInclude.length === 0) {
+      createPaymentResult = await Service.createPayments(name);
+      newUserPaymentId = createPaymentResult.insertId;
+    } else {
+      newUserPaymentId = checkInclude[0].id;
+    }
+    createUserPaymentResult = await Service.createUserPayments(userId, newUserPaymentId);
```
검증 부분 수정하신 것 확인 했습니다!

Zigje9 commented on client/src/components/molecules/CreditCard/CreditCard.tsx:
```diff
@@ -5,17 +5,31 @@ import CenterNormalText from '@atoms/p/CenterNormalText';
 
+const TextContainer = styled.div`
+  padding-top: 0.7rem;
+  flex: 5;
+`;
+
+const IconButtonContainer = styled.div`
+  flex: 1;
+`;
+
 const CreditCard: React.FC<Props> = ({ name }: Props) => {
   return (
     <ColumnFlexContainer width="100%">
       <RowFlexContainer width="100%" justifyContent="space-around">
-        <CenterNormalText>{name}</CenterNormalText>
-        <IconButton>{MinusIcon}</IconButton>
+        <TextContainer>
+          <CenterNormalText>{name}</CenterNormalText>
+        </TextContainer>
+        <IconButtonContainer>
+          <IconButton>{MinusIcon}</IconButton>
+        </IconButtonContainer>
       </RowFlexContainer>
       <Line widthPercent="110" margin="0 0.5rem 0 0" />
     </ColumnFlexContainer>
```
UI 가 훨씬 깔끔해진것 같네요~ 👍

leejiwoo2002 commented on client/src/components/organisms/CreditCardEditCardList/CreditCardEditCardList.tsx:
```diff
@@ -1,33 +1,10 @@
-    <ColumnFlexContainer width="100%" margin="2rem 0 0 0" justifyContent="space-between">
-      <ScrollDiv>
-        {cardNameList.map((cardName) => (
-          <CreditCard name={cardName} />
-        ))}
-      </ScrollDiv>
-    </ColumnFlexContainer>
-  );
+const CreditCardEditCardList: React.FC = () => {
+  return <></>;
```
아까 도연님이랑 말씀 하신게 이 부분인가 보네요!

JunYoung7 replied:
넵 추후 바로 수정하겠습니다!

Zigje9 reviewed (approved):
오늘하루도 늦게까지 고생하셨습니다 😁 👍

leejiwoo2002 reviewed (approved):
고생하셨습니다!

do02reen24 commented on client/src/components/organisms/CreditCardEditModal/CreditCardEditModal.tsx:
```diff
@@ -1,30 +1,40 @@
+    console.log(result);
+  };
+
+  const cardNameList = ['신한카드', '하나카드', '카카오페이', '신한카드', '하나카드', '카카오페이'];
```
이 부분은 Redux에서 불러와서 채워주는 것으로 수정가능할 것 같아요~ Redux에서 id도 함께 들고 있어서 나중에 삭제 구현하실 때 id 값만 넘겨주시면 될 것 같습니다😀 하다가 막히시는부분 잇으면 주말에도 호출해주세요~

JunYoung7 replied:
넵 나중에 받아와서 사용하도록 변경할 예정입니다. 든든한 도연님 믿고 열심히 구현하겠습니다. 😀

do02reen24 commented on server/src/payment/service.ts:
```diff
@@ -2,9 +2,27 @@ import 'dotenv/config';
+};
+
+export const getPayments = async () => {
+  const result = await sql(query.READ_PAYMENTS, []);
```
함수 사용여부 한 번 생각해보시고 나중에도 안쓰이는 것 같으면 삭제해주셔도 좋을 것 같아요 ㅎㅎ

JunYoung7 replied:
넵 사용 여지가 없다고 판단되면 삭제하겠습니다

do02reen24 reviewed (commented):
🍉수박🍉님 늦은시간까지 고생많으셨습니다😭💯👍
들어가서 푹쉬시고 내일 아침에 보죠😆🔥🔥🔥

do02reen24 commented:
POST response의 data 나중에 변경해주시면 좋을 것 같아요! 입력값이 여러개니까 반환 값도 여러개여야할 것 같습니당

JunYoung7 committed:
refactor : cardNameList 중복 수정


# <Why>를 위한 <What>
# 로그인을 위한 OAuth api 구현
# 객체 의존성 역전을 위한 프로토콜 추상화

# <타입> 리스트
#   feat    : 기능 (새로운 기능)
#   fix     : 버그 (버그 수정)
#   refactor: 리팩토링
#   style   : 스타일 (코드 형식, 세미콜론 추가: 비즈니스 로직에 변경 없음)
#   docs    : 문서 (문서 추가, 수정, 삭제)
#   test    : 테스트 (테스트 코드 추가, 수정, 삭제: 비즈니스 로직에 변경 없음)
#   chore   : 기타 변경사항 (빌드 스크립트 수정 등)
# --------------------------------------------------------------
# 메세지는 한글로 작성
# 타입 첫 문자는 소문자로 작성
# 제목의 끝은 명사형으로 작성
# 제목 끝에 마침표(.) 금지
# 본문은 "어떻게" 보다 "무엇을", "왜"를 설명
